### ISSUE TITLE
Handle Token File Names with Spaces Instead of Underscores

### ISSUE THREAD

**Issue by SabreRunner** [vtt-lair/token-replacer#49]
Please allow the module to read files as they are already saved instead of forcing a change to underscores.

**Comment by vtt-lair** [comment]
I'll see what I can do. I specifically did it this way to ensure better matching. If you are running under Windows, I've have batch files under each cr folder (called rename_files.bat) in the token folder that you can use to update your file names to replace spaces with underscores.

**Comment by SabreRunner** [comment]
It's not about renaming. It's about keeping things accessible and in specific order. I don't write English with underscores so I don't name my files with underscores. I don't even name my methods and fields with underscores.

**Comment by vtt-lair** [comment]
I've added functionality that will be release some time this week. In settings you can define what format is used for spaces in your file names. Currently there's 3 options, "Proper English", "Underscored" and "Dashed.

Hope this will help you out. Will comment here when the release is made, just want to run through a few scenarios still

**Comment by vtt-lair** [comment]
Some more details:

I've added some new basic functionality to define how the folder and image file names are defined. This way you can specify whether your folder and file names use spaces, underscores or dashes between the words.

So if you select the "Proper English" version for both folders and files, it'll be expecting "/cr 1/Monster Name.png".
If you select "Underscored" for both, it'll expect "cr_1/Monster_Name.png".
If you select "No Space" and "Underscored" for the folder and file names respectively (this is seen as default), it'll expect "/cr1/Monster_Name.png".

If you don't specify a NPC Difficulty name, it will ignore the folder and just compare things in the root folder.

_This will be released later this week_

**Comment by SabreRunner** [comment]
So I don't have to have them organised by CR? What if they are organised in another way?

**Comment by vtt-lair** [comment]
The way it currently works there's 2 ways you can organise the images.

The first is clearing out the NPC Difficulty Name from the Settings and putting all the images in the root folder of where you specified.
The second is having the NPC Difficulty Name specified in the Settings. This then gets used along with the NPC Difficulty Variable to discern where a specific monster/creature's image would be found.

If it's currently organised in another way, the way the modules works now would be difficult to take that into account. I wouldn't want to hard change it, as a few people are already using the structure (it was based on the way FA structures their tokens, just changed a but to take the way Foundry works with CR fractions into account).

I can maybe try and add an option where you specify to not use any specific structure and to just grab what is in folders and use just the name to grab the token. The obvious flaw with this would be that you might get some tokens with similar names and that could cause a wrong image to be shown for a token (which technically could happen when removing the NPC Difficulty Name and putting everything in the root).

**Comment by vtt-lair** [comment]
Ok. I've added a flag that specifies whether you use the NPC Difficulty to structure your folders against or not. It's enabled by default. Disabling it the only factor used to try and find the image for the token is the name.

I have added a warning though, that if for instance you have your own structure and we ignore the structure that if you for instance place a "Wolf", there a change it will use the "Wolf Spider". This is because I don't assume that your image names are always the full name, since I allow randomisation between "Wolf_Brown.png" and "Wolf_White.png", "Wolf_Spider_White.png" will be grab as a possibility of "Wolf" too.

I just want to run through more tests, but this will be released with the rest, later this week.
